geniusyield-market-maker/src/GeniusYield/MarketMaker/Prices.hs

```diff
@@ -65,6 +130,277 @@ data MMToken = MMToken
   deriving stock (Eq, Ord, Show, Generic)
   deriving (FromJSON, ToJSON) via CustomJSON '[FieldLabelModifier '[StripPrefix "mmt", LowerFirst]] MMToken
 
+data MMTokenPair = MMTokenPair
+  { mmtpCurrency  ∷ MMToken,
+    mmtpCommodity ∷ MMToken
+  }
+  deriving stock (Eq, Ord, Show)
+
+data MaestroPriceException
+  = MaestroPairNotFound
+  | MaestroApiError !Text !MaestroError
+  deriving stock (Show)
+  deriving anyclass (Exception)
+
+data MaestroPairOverride = MaestroPairOverride
+  { mpoPair             ∷ !String,
+    mpoCommodityIsFirst ∷ !Bool
+  }
+  deriving stock (Show, Generic)
+  deriving (FromJSON, ToJSON) via CustomJSON '[FieldLabelModifier '[CamelToSnake]] MaestroPairOverride
+
+data TaptoolsPairOverride = TaptoolsPairOverride
+  { ttpoAsset            ∷ !GYAssetClass,
+    ttpoPrecision        ∷ !Natural
+  }
+  deriving stock (Show, Generic)
+  deriving (FromJSON, ToJSON) via CustomJSON '[FieldLabelModifier '[CamelToSnake]] TaptoolsPairOverride
+
+data PricesProvidersAggregator = PPA
```

Comment:
Isn't this same as `PriceConfigV2`?

Reply:
Yes.  In my mind I was trying to separate the data types used for configuration and the ones used for constructing the `PricesProviders` with function `buildPP`.  But now that you point it out, there is no advantage in this strict separation.  Will use `PriceConfigV2` in place of `PricesProvidersAggregator`.

Reply:
`PPA` can be handled using pattern synonym.

Reply:
Managed to successfully work with a pattern synonym with the code:
```haskell
{-# COMPLETE PPA #-}
-- | Prices providers aggregator
pattern PPA :: PriceCommonCfg -> NonEmpty PricesProviderCfg -> PriceConfigV2
pattern PPA {ppaCommonCfg, ppaPricesCluster} = PriceConfigV2 { pcPriceCommonCfg = ppaCommonCfg, pcPricesProviderCfgs = ppaPricesCluster }
```

But in the end I introduced again datatype `PricesProvidersAggregator` to distinguish the data in `PricesProviders` (which is the result of `buildPP`) from the configuration data.  This was necessary to manage the MVar's that store the state of the mock prices providers in `PricesProviders`.